warehouse/encoding/jsonreader.go

```diff
@@ -7,46 +7,43 @@ import (
 	"io"
 )
 
-type JsonReader struct {
+type jsonReader struct {
 	scanner *bufio.Scanner
 }
 
-func (js *JsonReader) Read(columnNames []string) ([]string, error) {
-	var (
-		record []string
-		err    error
-	)
-
+func (js *jsonReader) Read(columnNames []string) ([]string, error) {
 	ok := js.scanner.Scan()
 	if !ok {
-		err = js.scanner.Err()
-		if err != nil {
+		if err := js.scanner.Err(); err != nil {
 			return []string{}, fmt.Errorf("scanner scan: %w", err)
 		}
 		return []string{}, io.EOF
 	}
 
 	lineBytes := js.scanner.Bytes()
-	jsonData := make(map[string]string)
+	jsonData := make(map[string]interface{})
```

Comment:
This came across while writing the tests. It's expecting only string which will not be the case.